### PR TITLE
AAE-24018 Readonly dropdown with the rest config don't fetch options when form rule change it to not readonly

### PR DESCRIPTION
**Pull request by wiktord2000**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [X] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [X] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
AAE-24018

https://github.com/user-attachments/assets/df688bc9-262e-476a-88a8-5abfb04dcb02

**Note:** We have form rule here: Enable dropdown when Text field has '1', otherwise is disabled. 

**What is the new behaviour?**

https://github.com/user-attachments/assets/9e7ae15c-a15c-44ae-b4f1-6c8615acd525

**Note:** We have form rule here: Enable dropdown when Text field has '1', otherwise is disabled. 

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by wiktord2000** [comment]
**Note:** With the current code is default to call api only when field is enabled by the applied rule (it would be ideal I guess), so the call is done at the form init even the dropdown is disabled.

**Comment by wiktord2000** [comment]
The actual issue refers to cloud widgets but I guess I have to update regular ones as well, I can remove the changes if I'am mistaken in **dropdown.widget.*** files

**Comment by kathrine0** [comment]
> **Note:** With the current code is difficult to call api only when field is not readonly by the applied rule (it would be ideal I guess), so the call is done at the form init even the dropdown is readonly at that time.

Can't we track the readonly value change and query the api only then? Why is it not possible?

**Comment by wiktord2000** [comment]
> > **Note:** With the current code is difficult to call api only when field is not readonly by the applied rule (it would be ideal I guess), so the call is done at the form init even the dropdown is readonly at that time.
> 
> Can't we track the readonly value change and query the api only then? Why is it not possible?

After small investigation, we relay on **@Input() field** from inherited component class -> [WidgetComponent](https://github.com/Alfresco/alfresco-ng2-components/blob/eda03ed525044e1b337d8c185eebeb09462d73cf/lib/core/src/lib/form/components/widgets/widget.component.ts#L52C1-L52C27), and readonly state is determined by **field.readOnly** property (when this rule will be fulfilled this value will change), I tried to use **ngOnChanges** hook on that but seems that we still relay on the same reference hence we can't track its changes. The benefit of current implementation is that we always call once when we will call every time readonly has changed it could lead to multiple api calls, which are not needed

File: lib/process-services/src/lib/form/widgets/dropdown/dropdown.widget.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export class DropdownWidgetComponent extends WidgetComponent implements OnInit {
     }
 
     ngOnInit() {
-        if (this.field?.restUrl) {
+        if (this.field?.restUrl && !this.field?.form?.readOnly) {
```
Comment (wiktord2000) [inline lib/process-services/src/lib/form/widgets/dropdown/dropdown.widget.ts:55]: Will move **this.field?.form?.readOnly** to the function as cloud one does

Comment (wiktord2000) [reply lib/process-services/src/lib/form/widgets/dropdown/dropdown.widget.ts:55]:
```
    private isReadOnlyForm(): boolean {
        return !!this.field?.form?.readOnly;
    }
```

**Review by kathrine0 — APPROVED** [review]

File: lib/process-services/src/lib/form/widgets/dropdown/dropdown.widget.ts (hunk 1 above)
Comment (wiktord2000) [reply lib/process-services/src/lib/form/widgets/dropdown/dropdown.widget.ts:55]: Done

**Pull request merged**